src/app/components/premium-badge.component.ts

```diff
@@ -1,17 +1,33 @@
-import { Component } from "@angular/core";
+import { Component, Input, OnInit } from "@angular/core";
 
 import { MessagingService } from "jslib-common/abstractions/messaging.service";
+import { StateService } from "jslib-common/abstractions/state.service";
 
 @Component({
   selector: "app-premium-badge",
   template: `
-    <a href="#" appStopClick *appNotPremium (click)="premiumRequired()">
-      <bit-badge type="success">{{ "premium" | i18n }}</bit-badge>
-    </a>
+    <button
+      bit-badge
+      badgeType="success"
+      *ngIf="condition"
+      appStopClick
+      (click)="premiumRequired()"
+      [attr.tabIndex]="tabindex"
+    >
+      {{ "premium" | i18n }}
+    </button>
   `,
 })
-export class PremiumBadgeComponent {
-  constructor(private messagingService: MessagingService) {}
+export class PremiumBadgeComponent implements OnInit {
+  // Optional condition defaults to premium
+  @Input() condition: boolean;
+  @Input() tabindex?: number = null;
```

Comment:
What's this variables purpose? Looks like we're always setting it to `-1` when using this the `report cards`